test/e2e/federated-secret.go

```diff
@@ -35,52 +36,111 @@ const (
 // Create/delete secret api objects
 var _ = framework.KubeDescribe("Federation secrets [Feature:Federation]", func() {
 	f := framework.NewDefaultFederatedFramework("federated-secret")
+	clusterClientSet := make(map[string]*release_1_3.Clientset)
 
 	Describe("Secret objects", func() {
-		AfterEach(func() {
+
+		BeforeEach(func() {
 			framework.SkipUnlessFederated(f.Client)
+			clusters := buildClustersOrFail_14(f)
+			for _, cluster := range clusters {
+				if _, found := clusterClientSet[cluster.Name]; !found {
+					clientset := createClientsetForCluster(*cluster, 1, "e2e-test")
+					clusterClientSet[cluster.Name] = clientset
+				}
+			}
+		})
 
+		AfterEach(func() {
+			framework.SkipUnlessFederated(f.Client)
 			nsName := f.FederationNamespace.Name
-			// Delete registered secrets.
-			// This is if a test failed, it should not affect other tests.
-			secretList, err := f.FederationClientset_1_4.Core().Secrets(nsName).List(api.ListOptions{})
-			Expect(err).NotTo(HaveOccurred())
-			for _, secret := range secretList.Items {
-				err := f.FederationClientset_1_4.Core().Secrets(nsName).Delete(secret.Name, &api.DeleteOptions{})
-				Expect(err).NotTo(HaveOccurred())
+			deleteAllTestSecrets(
+				f.FederationClientset_1_4.Core().Secrets(nsName).List,
+				f.FederationClientset_1_4.Core().Secrets(nsName).Delete,
+				false)
+			for _, clientset := range clusterClientSet {
+				deleteAllTestSecrets(
+					clientset.Core().Secrets(nsName).List,
+					clientset.Core().Secrets(nsName).Delete,
+					false)
```

Comment:
Ideally you need a call to unregisterClusters at the end of ForEach().  See 

https://github.com/kubernetes/kubernetes/blob/8c110675406ff822e7245563f9911fb173a0015e/test/e2e/federated-ingress.go#L101

and 

https://github.com/kubernetes/kubernetes/blob/4edc98260f26b2047ae9fcfa33cca1554dd1a3b2/test/e2e/federation-util.go#L163